internal/resource/registry.go

```diff
@@ -58,6 +77,25 @@ func (r *TypeRegistry) Register(registration Registration) {
 		panic(fmt.Sprintf("resource type %s already registered", key))
 	}
 
+	// set default acl hooks for those not provided
```

Comment:
I really like that you defaulted these instead of adding `nil` checks in the handlers 👏🏻